File: KingdeeWebAPI/KingdeeWebAPI.cpp
#include "pch.h"

#include "KingdeeWebAPI.h"
#include <string>  
#include <msclr/marshal_cppstd.h>

#ifdef _DEBUG
#using "..\\Debug\\KingdeeWebAPIdotNet.dll"
#else
#using "..\\Release\\KingdeeWebAPIdotNet.dll"
#endif // _DEBUG

using namespace msclr::interop;
using namespace KingdeeWebAPIdotNet;

void __stdcall Test()
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	kingdee->Test();
}

void __stdcall Login(const char* url, const char* dbid, const char* user, const char* pwd, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->Login(gcnew System::String(url), gcnew System::String(dbid), gcnew System::String(user), gcnew System::String(pwd));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

void __stdcall Logout()
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	kingdee->Logout();
}

void __stdcall Save(char* formid, char* data, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->Save(gcnew System::String(formid), gcnew System::String(data));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

void __stdcall Submit(char* formid, char* data, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->Submit(gcnew System::String(formid), gcnew System::String(data));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

void __stdcall Audit(char* formid, char* data, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->Audit(gcnew System::String(formid), gcnew System::String(data));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

void __stdcall UnAudit(char* formid, char* data, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->UnAudit(gcnew System::String(formid), gcnew System::String(data));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

void __stdcall Delete(char* formid, char* data, char* result)
{
	KingdeeWebAPIdotNet::KingdeeWebAPI^ kingdee = gcnew KingdeeWebAPIdotNet::KingdeeWebAPI();
	System::String^ ret = kingdee->Delete(gcnew System::String(formid), gcnew System::String(data));
	std::string szRet = marshal_as<std::string>(ret);
	strcpy(result, szRet.c_str());
}

File: KingdeeWebAPIdotNet/KingdeeWebAPI.cs
using Alex.Kingdee.Cloud.WebAPI.Client;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml;

namespace KingdeeWebAPIdotNet
{
    public class KingdeeWebAPI
    {
        

        public void Test()
        {
            K3CloudApiClient client = new K3CloudApiClient("http://fin.puruiit.cn:58888/K3Cloud/");
            var loginResult = client.ValidateLogin("5def4fed400b95", "倚和测试", "123456", 2052);
            Console.WriteLine(loginResult);
            client.Logout();
        }

        public void SetMode(int nMode)
        {
            Singleton.Instance.SetMode(nMode);
        }

        public string Login(string url, string dbid, string user, string pwd)
        {
            return Singleton.Instance.Login(url, dbid, user, pwd);
        }

        public void Logout()
        {
            Singleton.Instance.Logout();
        }

        public string Save(string formid, string data)
        {
            return Singleton.Instance.Save(formid, data);
        }

        public string Submit(string formid, string data)
        {
            return Singleton.Instance.Submit(formid, data);
        }

        public string Audit(string formid, string data)
        {
            return Singleton.Instance.Audit(formid, data);
        }

        public string UnAudit(string formid, string data)
        {
            return Singleton.Instance.UnAudit(formid, data);
        }

        public string Delete(string formid, string data)
        {
            return Singleton.Instance.Delete(formid, data);
        }

        
    }
}
